arbor/morph/region.cpp

```diff
@@ -306,9 +307,329 @@ std::ostream& operator<<(std::ostream& o, const all_& t) {
     return o << "(all)";
 }
 
+// Region with all segments distal from another region
 
-// Named region.
+struct distal_interval_ {
+    locset start;
+    double distance; //um
+};
+
+region distal_interval(locset start, double distance) {
+    return region(distal_interval_{start, distance});
+}
+
+mcable_list thingify_(const distal_interval_& reg, const mprovider& p) {
+    const auto& m = p.morphology();
+    const auto& e = p.embedding();
+
+    std::vector<mcable> L;
+
+    auto start = thingify(reg.start, p);
+    auto distance = reg.distance;
+
+    struct branch_interval {
+        msize_t bid;
+        double distance;
+    };
+
+    for (auto c: start) {
+        std::stack<branch_interval> branches_reached;
+        bool first_branch = true;
+
+        // Transform end point of branch into the start point of its parent
+        if (c.pos == 0) {
+            c = {m.branch_parent(c.branch),1};
+        }
+
+        // if we're starting at the end of a branch, start traversal with its children
+        if (c.pos < 1) {
+            branches_reached.push({c.branch, distance});
+        } else {
+            first_branch = false;
+            for (auto child: m.branch_children(c.branch)) {
+                branches_reached.push({child, distance});
+            }
+        }
+
+        while (!branches_reached.empty()) {
+            auto bi = branches_reached.top();
+            branches_reached.pop();
+
+            auto branch = bi.bid;
+            auto rem_dist = bi.distance;
+
+            auto branch_length = e.branch_length(branch);
+            auto prox_pos = first_branch*c.pos;
+            auto dist_pos = rem_dist / branch_length + prox_pos;
+
+            if (dist_pos <= 1) {
+                L.push_back({branch, prox_pos, dist_pos});
+            } else {
+                L.push_back({branch, prox_pos, 1});
+                rem_dist = rem_dist - (1 - prox_pos)*branch_length;
+                for (auto child: m.branch_children(branch)) {
+                    branches_reached.push({child, rem_dist});
+                }
+            }
+            first_branch = false;
+        }
+    }
+    util::sort(L);
+    return merge(L);
+}
```

Comment:
If a distance is zero and locations of the form (bid, 0) are given, there will be zero-length cables (bid, 0, 0) in the list L which will need to be merged/canonified. `remove_cover` will do this and also the sort and merge. (Though I think in the future we should re-examine the cover/remove_cover/merge subroutines to make the semantics of all this a bit clearer.)